aim/web/api/runs/views.py

```diff
@@ -481,3 +482,34 @@ async def remove_run_tag_api(run_id: str, tag_id: str, factory=Depends(object_fa
         'removed': removed,
         'status': 'OK'
     }
+
+
+@runs_router.delete('/{run_id}/')
+async def delete_run_api(run_id: str):
+    # Get project
+    project = Project()
+    if not project.exists():
+        raise HTTPException(status_code=404)
+    success = project.repo.delete_run(run_id)
+    if not success:
+        raise HTTPException(400, detail=f'Error while deleting run {run_id}.')
+
+    return {
+        'status': 'OK'
+    }
+
+
+@runs_router.post('/delete-batch/')
+async def delete_runs_batch_api(runs_batch: RunsBatchIn):
+    # Get project
+    project = Project()
+    if not project.exists():
+        raise HTTPException(status_code=404)
```

Comment:
we use this part in so many places maybe its time to wrap int under a decorator?

Reply:
agree @devfox-se. I do have this fixed in a different branch (for API unification).